Sources/Brave/Frontend/UserContent/UserScripts/Scripts_Dynamic/Scripts/Paged/WalletEthereumProviderScript.js

```diff
@@ -30,24 +30,58 @@ if (window.isSecureContext) {
     }));
   }
   
-  const provider = {
-    value: {
-      chainId: undefined,
-      networkVersion: undefined,
-      selectedAddress: undefined,
-      isMetaMask: true,
-      isBraveWallet: true,
-      request: $(function (args) /* -> Promise<unknown> */  {
+  var EventEmitter = require('events');
+  var BraveWeb3ProviderEventEmitter = new EventEmitter();
+  
+  const provider = {value: {}};
+  $Object.defineProperty(window, 'ethereum', provider);
+  $Object.defineProperty(window, 'braveEthereum', provider);
```

Comment:
can a page script redefine window.ethereum itself? like by doing `delete window.ethereum` and then setting it to be `{isBraveWallet: false}`

also if the subproperties are configurable, i think page scripts can overwrite them by deleting and redefining them too

Reply:
@diracdeltas `delete window.ethereum` fails because `configurable` is false by default in `Object.defineProperty`. Similarly with `Object.defineProperties`, `configurable` is false by default as well so the subproperties cannot be deleted either.

<img width="950" alt="Screenshot 2023-09-06 at 7 04 47 PM" src="https://github.com/brave/brave-ios/assets/5314553/4a5c3099-6f10-428d-9196-add20bc52c0e">